packages/gatsby-plugin-sitemap/src/gatsby-node.js

```diff
@@ -26,7 +40,37 @@ exports.onPostBuild = async ({ graphql, pathPrefix }, pluginOptions) => {
     excludeOptions,
     pathPrefix
   )
-  serialize(queryRecords).forEach(u => map.add(u))
+  const urls = serialize(queryRecords)
 
-  return await writeFile(saved, map.toString())
+  if (!(rest.sitemapSize && urls.length > rest.sitemapSize)) {
+    const map = sitemap.createSitemap(rest)
+    serialize(queryRecords).forEach(u => map.add(u))
+    return await writeFile(saved, map.toString())
+  } else {
+    const {
+      site: {
+        siteMetadata: { siteUrl },
+      },
+    } = queryRecords
+    return new Promise((resolve, reject) => {
+      // sitemapv-index.xml is default file name. (https://git.io/fhNgG)
+      const indexFilePath = path.join(
+        distDir,
+        `${rest.sitemapName || `sitemap`}-index.xml`
+      )
+      const sitemapIndexOptions = {
+        ...rest,
+        ...{
+          hostname: hostname || withoutTrailingSlash(siteUrl),
+          targetFolder: distDir,
+          urls,
+          callback: error => {
+            if (error) reject()
+            renameFile(indexFilePath, saved).then(resolve)
+          },
+        },
```

Comment:
```suggestion
        },
```
same here as we remove line `...{`